File: adventjs/2025/02/solution.py
def manufacture_gifts(gifts_to_produce):
  result = []
  for gift in gifts_to_produce:
    quantity = gift.get('quantity', 0)
    if isinstance(quantity, (int, float)) and quantity > 0:
      for _ in range(int(quantity)):
        result.append(gift.get('toy', ''))
  return result

File: adventjs/2025/02/solution.ts
function manufactureGifts(
  giftsToProduce: Array<{ toy: string; quantity: number }>
): string[] {
  const result: string[] = []
  for (const gift of giftsToProduce) {
    if (typeof gift.quantity === 'number' && gift.quantity > 0) {
      for (let i = 0; i < gift.quantity; i++) {
        result.push(gift.toy)
      }
    }
  }
  return result
}
